### PR TITLE
Emit `mtu-exceeded` event when write first exceeds MTU

### PR DESCRIPTION
**Pull request by kasperisager**

### CODE REVIEW

File: lib/stream.js [hunk 1]
```diff
@@ -183,6 +187,10 @@ module.exports = class UDXStream extends streamx.Duplex {
 
     if (drained) cb(null)
     else this._onwrite = cb
+
+    if (buffer.byteLength > this.mtu - 20 /* header */) {
+      this.emit('fragmentation')
```
Comment (mafintosh) [inline lib/stream.js:192]: should we only emit this once ever?

Comment (kasperisager) [reply lib/stream.js:192]: I guess we should find another event name if we do that. Callers can also always do `stream.once('fragmentation')`.

Comment (mafintosh) [reply lib/stream.js:192]: is just emitted a lot, always a sucker for reducing overhead 😬

Comment (mafintosh) [reply lib/stream.js:192]: (i remember in streamx the data event was a bottleneck)

Comment (mafintosh) [reply lib/stream.js:192]: what about `mtu-exceeded` and then we do it once?

Comment (kasperisager) [reply lib/stream.js]: All set 👌

**Comment by mafintosh** [comment]
Looks great. @kasperisager ready to go from you?

**Comment by kasperisager** [comment]
@mafintosh Yep, all set 👍

**Pull request merged**